src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -219,6 +221,11 @@ public void performSearch() {
             return;
         }
 
+        if (regularExp.isSelected() && !validRegex()) {
+            currentResults.setText(Localization.lang("Invalid Regex"));
```

Comment:
As @Siedlerchr suggests, it would be nice if you could report the invalid regex to the user directly in the text box. We use the following validation framework for this: https://github.com/sialcasa/mvvmFX/wiki/Validation (the other places where this library is used might give you a good idea how to implement it - sounds more scarier than it actually is)